test/model/autocomplete_test.dart

```diff
@@ -305,40 +305,49 @@ void main() {
-    doCheck('Fully Named', eg.user(fullName: 'Full Name'), false);
-    doCheck('Full Name', eg.user(fullName: 'Full'), false);
-    doCheck('Full Name', eg.user(fullName: 'Name'), false);
-    doCheck('ull ame', eg.user(fullName: 'Full Name'), false);
-    doCheck('ull Name', eg.user(fullName: 'Full Name'), false);
-    doCheck('Full ame', eg.user(fullName: 'Full Name'), false);
-    doCheck('Full Full', eg.user(fullName: 'Full Name'), false);
-    doCheck('Name Name', eg.user(fullName: 'Full Name'), false);
-    doCheck('Name Full', eg.user(fullName: 'Full Name'), false);
-    doCheck('Name Four Full Words', eg.user(fullName: 'Full Name Four Words'), false);
-    doCheck('F Full', eg.user(fullName: 'Full Name Four Words'), false);
-    doCheck('Four F', eg.user(fullName: 'Full Name Four Words'), false);
+    test('user is included if fullname words match the query', () {
+      doCheck('', eg.user(fullName: 'Full Name'), true);
+      doCheck('', eg.user(fullName: ''), true); // Unlikely case, but should not crash
+      doCheck('Full Name', eg.user(fullName: 'Full Name'), true);
+      doCheck('full name', eg.user(fullName: 'Full Name'), true);
+      doCheck('Full Name', eg.user(fullName: 'full name'), true);
+      doCheck('Full', eg.user(fullName: 'Full Name'), true);
+      doCheck('Name', eg.user(fullName: 'Full Name'), true);
+      doCheck('Full Name', eg.user(fullName: 'Fully Named'), true);
+      doCheck('Full Four', eg.user(fullName: 'Full Name Four Words'), true);
+      doCheck('Name Words', eg.user(fullName: 'Full Name Four Words'), true);
+      doCheck('Full F', eg.user(fullName: 'Full Name Four Words'), true);
+      doCheck('F Four', eg.user(fullName: 'Full Name Four Words'), true);
+      doCheck('full full', eg.user(fullName: 'Full Full Name'), true);
+      doCheck('full full', eg.user(fullName: 'Full Name Full'), true);
+
+      doCheck('F', eg.user(fullName: ''), false); // Unlikely case, but should not crash
+      doCheck('Fully Named', eg.user(fullName: 'Full Name'), false);
+      doCheck('Full Name', eg.user(fullName: 'Full'), false);
+      doCheck('Full Name', eg.user(fullName: 'Name'), false);
+      doCheck('ull ame', eg.user(fullName: 'Full Name'), false);
+      doCheck('ull Name', eg.user(fullName: 'Full Name'), false);
+      doCheck('Full ame', eg.user(fullName: 'Full Name'), false);
+      doCheck('Full Full', eg.user(fullName: 'Full Name'), false);
+      doCheck('Name Name', eg.user(fullName: 'Full Name'), false);
+      doCheck('Name Full', eg.user(fullName: 'Full Name'), false);
+      doCheck('Name Four Full Words', eg.user(fullName: 'Full Name Four Words'), false);
+      doCheck('F Full', eg.user(fullName: 'Full Name Four Words'), false);
+      doCheck('Four F', eg.user(fullName: 'Full Name Four Words'), false);
+    });
+
+    test('user is always excluded when not active regardless of other criteria', () {
+      doCheck('Full Name', eg.user(fullName: 'Full Name', isActive: false), false);
```

Comment:
Just to make this test case fully self-contained, let's give it a second line that looks exactly the same except the user is active, and does match.

That line will be identical to one of the lines in the existing test case, so in a sense it's redundant. But it makes a useful baseline for the reader to compare this other check to. It means that the reader can look at this test case and be confident that it really is testing what it's meant to test — that there isn't some accident where the user stopped matching because something changed about the name matching, or whatever — without having to look around at other tests, or to look back at the implementation that's being tested.